package/Dockerfile

```diff
@@ -235,61 +299,68 @@ ENV ETCD_UNSUPPORTED_ARCH ${ETCD_UNSUPPORTED_ARCH}
 
 
 # Enable exporting of the k3s images as part of the build process.
-FROM --platform=$BUILDPLATFORM registry.suse.com/bci/golang:1.24 AS images
-
-ENV GOMODCACHE=/root/.cache/go/modcache
-ENV GOCACHE=/root/.cache/go/cache
-
+FROM --platform=$BUILDPLATFORM go-builder AS images
+WORKDIR /src
+ENV CGO_ENABLED=1
 ENV CATTLE_K3S_VERSION v1.32.1+k3s1
-
 RUN zypper -n install libbtrfs-devel libgpgme-devel
-
-WORKDIR /src
 COPY hack/airgap/go.mod hack/airgap/go.sum /src/
 RUN --mount=type=cache,target=/root/.cache,id=rancher go mod download
 COPY hack/airgap/ /src/
-
 RUN --mount=type=cache,target=/root/.cache,id=rancher go build -tags k3s_export -o export-images ./...
 RUN ./export-images -k3s-version ${CATTLE_K3S_VERSION} -output /src/k3s-airgap-images.tar
 
 
-FROM --platform=$BUILDPLATFORM registry.suse.com/bci/golang:1.24 AS build
+# rancher-go-builder is an intermediate stage with go mod ready and Rancher
+# dependencies downloaded. Can be used to build both the Rancher server and agent
+# binaries.
+FROM --platform=$BUILDPLATFORM go-builder AS rancher-go-builder
+WORKDIR /app
+# Only invalidate cache if go.mod/go.sum changes.
+COPY go.mod go.sum .
+COPY pkg/apis/go.mod pkg/apis/go.sum pkg/apis/
+COPY pkg/client/go.mod pkg/client/go.sum pkg/client/
+RUN --mount=type=cache,target=/root/.cache,id=rancher go mod download
+RUN --mount=type=cache,target=/root/.cache,id=rancher cd pkg/apis && go mod download
+RUN --mount=type=cache,target=/root/.cache,id=rancher cd pkg/client && go mod download
+
+
+FROM --platform=$BUILDPLATFORM rancher-go-builder AS server-build
 ARG VERSION
 ARG COMMIT
 ARG RKE_VERSION
-ARG CGO_ENABLED=0
```

Comment:
@tashima42 since we're on the topic of CGO, we can see here there was an `ARG CGO_ENABLED=0` but no variables used and no `ENV` directive either. So it feels like CGO was enabled before. My PR does add `ENV CGO_ENABLED=$CGO_ENABLED` though so now CGO should definitely be disabled when building the go binary.

Do you see this as an issue?